pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,384 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/route/api"
+	fileutil "github.com/openshift/origin/pkg/util/file"
+)
+
+// NewCmdCreateRoute is a macro command to create a new route.
+func NewCmdCreateRoute(f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "route",
+		Short: "Create a secure route using specified subcommand.",
+		Long:  "Create a secure route using specified subcommand.",
```

Comment:
Longer long desc, you should probably describe what's supported like "secure routes support 3 TLS termination strategies:\n 1) ..." and so on.